pisp_app/lib/route_generator.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:flutter/material.dart';
+import 'package:pispapp/Screens/dashboard.dart';
+
+// To encapsulate logic around routing and route guards at one place
+class RouteGenerator {
+  static Route<dynamic> generateRoute(RouteSettings settings) {
+    // Getting arguments passed in while calling Navigator.pushNamed
+    final Object args = settings.arguments;
+
+    switch (settings.name) {
+      case '/':
+        return MaterialPageRoute<dynamic>(builder: (_) => Dashboard());
+      case '/second':
+        // Validation of correct data type
+        if (args is String) {
+          return MaterialPageRoute<dynamic>(
+            builder: (_) => Dashboard(
```

Comment:
The idea was to put all routing logic in one file eventually. Currently I have Navigator.push all over in the codebase. I just setup the skeleton to remove all of this from the codebase later, and use named routes with the switch case inside one function. Does that make sense ?

Reply:
Ah I see, okay got it, I think it is okay if you want to put all the routing in one place.
But, I think you still need to pass `setting` when instantiating `MaterialPageRoute` to avoid the mentioned problem.

Reply:
@MahidharBandaru I think this hasn't been updated?

Reply:
https://github.com/flutter/flutter/issues/50192#issuecomment-590307455